utils/http/retry_test.go

```diff
@@ -0,0 +1,94 @@
+package http
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/atomic"
+
+	"github.com/ARM-software/golang-utils/utils/commonerrors"
+	"github.com/ARM-software/golang-utils/utils/commonerrors/errortest"
+	"github.com/ARM-software/golang-utils/utils/logs/logstest"
+)
+
+func TestRetryOnError(t *testing.T) {
+	tests := []struct {
+		policy *RetryPolicyConfiguration
+	}{
+		{
```

Comment:
can we test nil? so that if the implementation is changed in the future and someone forgets to account for nil, we can catch it